### PR TITLE
Update to Java 11,  react-native-ble-plx 3.1.2

### PR DESCRIPTION
**Pull request by jmbrunskill**
This PR Updates the  react-native-ble-plx library.

The updated library might help with data collection reliability issues such as #251 (Temperature Spikes)
It also paves the way for reading the advertisement packets as it provides access to the Raw scan data.

Inorder to get the new version working I had to use a newer version of Java and target a newer version of the Android SDK. This seems like a bit of a shame in some ways but also not a bad thing?

A POC of reading this has been done, example code here...

```

const rawBuffer = Buffer.from(device.rawScanRecord, 'base64')

interface ParsedData {
  battery?: number
  logIntevalSec?: number
  storedLogCount?: number
  currentTemp?: number
  currentHumidity?: number
}

const parseData = (rawScanRecord: Buffer) => {
  if (rawScanRecord.length < 16) {
    return null
  }
  const version = rawScanRecord.readUInt8(7)
  // console.log('version:', version)
  if (version !== 23) {
    // Unsupported version
    return null
  }

  const battery = rawScanRecord.readUInt8(8)

  const logIntevalSec = rawScanRecord.readUInt16BE(9)

  const storedLogCount = rawScanRecord.readUInt16BE(11)

  const currentTemp = rawScanRecord.readUInt16BE(13) / 10.0

  const currentHumidity = rawScanRecord.readUInt16BE(15) / 10.0

  return { battery, logIntevalSec, storedLogCount, currentTemp, currentHumidity }
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,11 +22,11 @@ See [https://docs.msupply.foundation/en:cold_chain:start](https://docs.msupply.f
...
 - Install SDKMAN for managing Java versions: https://sdkman.io/.
 
-The application is using java v1.8 you may need to target a specific java version in order to run the app locally.
-e.g.
+The application is using java v11 you may need to target a specific java version in order to run the app locally.
+e.g. (with sdkman installed):
 
 ```
-export JDK_HOME=~/.jenv/versions/oracle64-1.8.0.192 && JAVA_HOME=~/.jenv/versions/oracle64-1.8.0.192 && yarn start
+ sdk use java 11.0.23-zulu
```
Comment (jmbrunskill) [inline README.md:29]: Added this as it seems we need java11+ now that we're targeting a SDK 23.
Haven't tired 17 or other more modern ones, wasn't working before...

File: android/app/src/main/java/com/msupplycoldchain/SchedulerService.java [hunk 2]
```diff
@@ -63,7 +63,16 @@ public void onDestroy() {
 
     public static Notification buildNotification(Context context, String text) {
         Intent notificationIntent = new Intent(context, MainActivity.class);
-        PendingIntent contentIntent = PendingIntent.getActivity(context, 0, notificationIntent, PendingIntent.FLAG_CANCEL_CURRENT);
+        PendingIntent contentIntent = null;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S) {
+            contentIntent = PendingIntent.getActivity
+                (context, 0, notificationIntent, PendingIntent.FLAG_MUTABLE);
```
Comment (jmbrunskill) [inline android/app/src/main/java/com/msupplycoldchain/SchedulerService.java:69]: Apparenlty IMMUTABLE Is safer but wanted to just get it working for now...

File: README.md (hunk 1 above)
Comment (mark-prins) [reply README.md:29]: api level 23 is [android 6](https://developer.android.com/tools/releases/platforms#6.0) - given that the samsung I'm using is older, and android 12.. we're probably ok

File: .yalc/msupply-ble-service/package.json [hunk 3]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "msupply-ble-service",
-  "version": "0.5.3",
+  "version": "0.5.1",
```
Comment (mark-prins) [inline .yalc/msupply-ble-service/package.json:3]: why would the version decrease?

File: .yalc/msupply-ble-service/lib/typescript/constants/index.d.ts [hunk 4]
```diff
@@ -1,3 +1,2 @@
 export { BT510 } from './Bluetooth';
 export { BLUE_MAESTRO } from './Bluetooth';
-export { MILLISECONDS } from './Milliseconds';
```
Comment (mark-prins) [inline .yalc/msupply-ble-service/lib/typescript/constants/index.d.ts:3]: You could delete the MILLISECONDS const altogether if not used

File: .yalc/msupply-ble-service/src/Bluetooth/BleService.ts [hunk 5]
```diff
@@ -55,67 +55,42 @@ const dummyLogger: Logger = {
...
-      this.logger.debug(`${deviceDescriptor} Connecting to other device`);
-      const deviceIsConnected = await this.manager.isDeviceConnected(device.id);
-      if (deviceIsConnected) {
-        this.logger.debug(`${deviceDescriptor} Disconnecting`);
-        await this.manager.cancelDeviceConnection(device.id);
-      }
+    const deviceIsConnected = await this.manager.isDeviceConnected(device.id);
+    this.logger.info('deviceIsConnected?', { deviceIsConnected });
+    if (deviceIsConnected) {
+      await this.manager.cancelDeviceConnection(device.id);
```
Comment (mark-prins) [inline .yalc/msupply-ble-service/src/Bluetooth/BleService.ts:85]: was this deliberate? It looks like an older version has snuck in

**Review by mark-prins — COMMENTED** [review]
App is running nicely - so I approve that part. Am wondering though if an older version of the ble-service is being used? can you check / revert those changes? shouldn't affect the upgrade aspec.

File: .yalc/msupply-ble-service/package.json (hunk 3 above)
Comment (jmbrunskill) [reply .yalc/msupply-ble-service/package.json:3]: I wonder if I branched off main but we build from develop or something. Will take a quick look at this tomorrow.

Comment (jmbrunskill) [reply .yalc/msupply-ble-service/package.json]: It was this, now updated from main

**Review by mark-prins — APPROVED** [review]
Thanks @jmbrunskill - Looks good!

**Pull request merged**